variantlib/models/configuration.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from dataclasses import field
+
+from variantlib.constants import VALIDATION_KEY_REGEX
+from variantlib.constants import VALIDATION_NAMESPACE_REGEX
+from variantlib.models.base import BaseModel
+from variantlib.models.validators import validate_and
+from variantlib.models.validators import validate_instance_of
+from variantlib.models.validators import validate_list_matches_re
+from variantlib.models.validators import validate_list_of
+
+
+@dataclass(frozen=True)
+class Configuration(BaseModel):
+    """
+    Configuration class for variantlib.
+
+    This class is used to define the configuration for the variantlib library.
+    It includes fields for the namespace, key, and value, along with validation
+    checks for each field.
+
+    Attributes:
+        namespaces_by_priority (list): The namespace of the configuration.
+        features_by_priority (list): The key of the configuration.
```

Comment:
That seems to mismatch our vocabulary of naming it a "key".

Reply:
Actually that file shouldn't have gone in this PR. I'll remove it

Reply:
Done